### PR TITLE
Support Native Ghost Widgets

### LINKED ISSUE #16: Warning on `connect.session()`
`express-session` creates a warning on a `production` environment - 
```shell
Warning: connect.session() MemoryStore is not designed for a production environment, as it will leak memory, and will not scale past a single process.
```

The alternative might be using `cookie-session`.

---
Reference - https://stackoverflow.com/a/59629154/6819340

### PR DESCRIPTION
**Pull request by ItzNotABug**
This PR adds support for more Ghost widgets (Cards) to be used in the newsletter. 
Currently added -
- Bookmark
- Native Button
- Toggle
- Pre x Code
- Audio, Video, File
- Images / Unsplash
- YouTube
- Twitter (X)

---
### Screenshots -
As taken on Gmail iOS App

<table>
  <tr>
    <td>
      Ghosler Widget - 1
    </td>
    <td>
      Ghosler Widget - 2
    </td>
    <td>
      Ghosler Widget - 3
    </td>
  </tr>
  <tr>
    <td>
      Ghosler Widget - 4
    </td>
    <td>
      Ghosler Widget - 5
    </td>
    <td>
      Ghosler Widget - 6
    </td>
  <tr>
</table>

### CODE REVIEW

**Comment by ItzNotABug** [comment]
Remaining widgets -
1. Vimeo
2. Codepen
3. Spotify 
4. Soundcloud

**Comment by ItzNotABug** [comment]
Spotify, Soundcloud cannot be added because they use an `iframe` to render content.
CodePen failed to load all sort of content for me so I could not parse its layout as it wasn't available.

**Comment by ItzNotABug** [comment]
Also this PR contains the fix for #16 so the `cookie-session` branch can be deleted.

File: utils/newsletter.js [hunk 1]
```diff
@@ -139,11 +141,14 @@ export default class Newsletter {
...
+            const injectedHtml = injectUrlTracking
                 ? await this.#injectUrlTracking(renderingData, template)
-                : {trackedLinks: [], modifiedHtml: template};
+                : {trackedLinks: new Set(), modifiedHtml: template};
+
+            // add widgets, inline css and minify the html.
+            return await Widgets.replace(renderingData, injectedHtml);
         } catch (error) {
-            logError(logTags.Newsletter, error);
+            logError(logTags.Newsletter, error.message);
```
Comment (ItzNotABug) [inline utils/newsletter.js]: This needs to be `error` only. Seems like I missed changing it back post some testing.

File: utils/widgets.js [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+            const imageUrl = img.attr('src');
+
+            if (imageUrl) {
+                const currentStyle = $(this).attr('style') || '';
+                $(this).attr('style', `${currentStyle} background-image: url('${imageUrl}');`);
+            }
+        });
+
+        $('.kg-bookmark-container').each(function () {
+            $(this).attr('style', 'border-radius: 5px; overflow:hidden;');
```
Comment (ItzNotABug) [inline utils/widgets.js]: I think we should probably try adding this in the `<style>` tag.

File: views/dashboard/details.ejs [hunk 3]
```diff
@@ -94,6 +94,12 @@
         <p class="font-semibold text-center text-gray-400 my-6">URLs are tracked and incremented<br>every time they are
             clicked.</p>
 
+        <% post.stats.postContentTrackedLinks.sort((a, b) => {
+            const clicksA = Object.values(a)[0];
+            const clicksB = Object.values(b)[0];
+            return clicksB - clicksA;
```
Comment (ItzNotABug) [inline views/dashboard/details.ejs]: Maybe use a one liner here?

**Review by ItzNotABug — COMMENTED** [review]
Minor changes.

**Comment by ItzNotABug** [comment]
Added the screenshots to the description for better clarity.

**Comment by viriatusX** [comment]
Incredible 😍👏

**Pull request merged**